src/js/ripple/remote.js

```diff
@@ -9,117 +10,72 @@
-var Transaction      = require('./transaction').Transaction;
-var Account          = require('./account').Account;
-var Meta             = require('./meta').Meta;
-var OrderBook        = require('./orderbook').OrderBook;
-var PathFind         = require('./pathfind').PathFind;
+
+var EventEmitter = require('events').EventEmitter;
+var util = require('util');
+var assert = require('assert');
+var LRU = require('lru-cache');
+var async = require('async');
+var lodash = require('lodash');
+var Server = require('./server').Server;
+var Request = require('./request').Request;
+var Server = require('./server').Server;
+var Amount = require('./amount').Amount;
+var Currency = require('./currency').Currency;
+var UInt160 = require('./uint160').UInt160;
+var UInt256 = require('./uint256').UInt256;
+var Transaction = require('./transaction').Transaction;
+var Account = require('./account').Account;
+var Meta = require('./meta').Meta;
+var OrderBook = require('./orderbook').OrderBook;
+var PathFind = require('./pathfind').PathFind;
 var SerializedObject = require('./serializedobject').SerializedObject;
-var RippleError      = require('./rippleerror').RippleError;
-var utils            = require('./utils');
-var sjcl             = require('./utils').sjcl;
-var hashprefixes     = require('./hashprefixes');
-var config           = require('./config');
-var log              = require('./log').internal.sub('remote');
-
-/**
- *    Interface to manage the connection to a Ripple server.
- *
- *    This implementation uses WebSockets.
- *
- *    Keys for opts:
- *
- *      trace
- *      max_listeners      : Set maxListeners for remote; prevents EventEmitter warnings
- *      connection_offset  : Connect to remote servers on supplied interval (in seconds)
- *      trusted            : truthy, if remote is trusted
- *      max_fee            : Maximum acceptable transaction fee
- *      fee_cushion        : Extra fee multiplier to account for async fee changes.
- *      servers            : Array of server objects with the following form
- *      canonical_signing  : Signatures should be canonicalized and the "canonical" flag set
- *
- *         {
- *              host:    <string>
- *            , port:    <number>
- *            , secure:  <boolean>
- *         }
- *
- *    Events:
- *      'connect'
- *      'disconnect'
- *      'state':
- *      - 'online'        : Connected and subscribed.
- *      - 'offline'       : Not subscribed or not connected.
- *      'subscribed'      : This indicates stand-alone is available.
- *
- *    Server events:
- *      'ledger_closed'   : A good indicate of ready to serve.
- *      'transaction'     : Transactions we receive based on current subscriptions.
- *      'transaction_all' : Listening triggers a subscribe to all transactions
- *                          globally in the network.
- *
- *    @param opts      Connection options.
- *    @param trace
- */
-
-function Remote(opts, trace) {
```

Comment:
Most of this comment block is documented [here](https://github.com/ripple/ripple-lib/blob/develop/docs/REFERENCE.md#remote-options). I might be convinced to add the old comment block back, but it's very unlikely that someone else's code comments are going to be maintained, as we've seen. IMO it adds more clutter than utility, but I share your concern for documentation.